src/lib.rs

```diff
@@ -62,5 +62,25 @@ macro_rules! fuzz_target {
 
             $body
         }
+
+
+        #[export_name = "LLVMFuzzerCustomOutput"]
+        pub extern "C" fn output(ptr: *const u8, len: usize) {
+            use arbitrary::{Arbitrary, RingBuffer};
+
+            let bytes = unsafe { std::slice::from_raw_parts(ptr, len) };
+
+            let mut buf = match RingBuffer::new(bytes, bytes.len()) {
+                Ok(b) => b,
+                Err(_) => return,
+            };
+
+            let data: $dty = match Arbitrary::arbitrary(&mut buf) {
+                Ok(d) => d,
+                Err(_) => return,
```

Comment:
These error cases should probably fall back to the default outputter, and print the bytes or whatever, right?

Reply:
Ah I see that they will, since this is *additional* output, not replacing the existing output.